### ISSUE TITLE
Route does not find a match when using parenthesis in URL

### ISSUE THREAD

**Issue by VGRAJED** [SAP/spartacus#13559]
Using Spartacus 3.4 (Angular 10).

### Example

Works: https://spartacus-demo.eastus.cloudapp.azure.com/electronics-spa/en/USD/product/514518/ACK-E2
Also works: https://spartacus-demo.eastus.cloudapp.azure.com/electronics-spa/en/USD/product/514518/ACK-E2testestestest

Does not work: https://spartacus-demo.eastus.cloudapp.azure.com/electronics-spa/en/USD/product/514518/ACK-E2(hello)

The issue can be reproduced by using a parenthesis anywhere in the URL.

**Comment by Xymmer** [comment]
Let's investigate what we can do to allow () but also check what Commerce Cloud also actually supports. Maybe there is a reason why () is not supported.

**Comment by Platonn** [comment]
**TLDR: It's Angular's feature / limitation, which can be workarounded in various ways. Not a Spartacus bug.**

By default, Angular [treats parenthesis specially](https://angular.io/api/router/DefaultUrlSerializer#description) (for defining [_secondary router outlets_](https://angular.io/guide/router-tutorial-toh#secondary-routes)). Therefore for any typical parenthesis you want to place in the URL, please URI-encode them first.

`(` -> `%28`
`)` -> `%29`

a) If your URLs are already defined in the CMS with raw parenthesis, you might need to change them manually in your CMS data.
b) If you're just creating links in Angular/Spartacus (with the `[routerLink]` directive), Angular will URI-encode any special characters (including parenthesis) for you by default. 

If you really want Angular to treat raw parenthesis normally, you would need to provide a custom `UrlSerializer`, i.e. by extending and customizing the Angular class [`DefaultUrlSerializer`](https://angular.io/api/router/DefaultUrlSerializer). However, this might be a bit cumbersome, because [Spartacus already provides `UrlSerializer`](https://github.com/SAP/spartacus/blob/release/4.1.x/projects/core/src/site-context/providers/site-context-params-providers.ts#L11) by [extending `DefaultUrlSerializer` in `SiteContextUrlSerializer`](https://github.com/SAP/spartacus/blob/release/4.1.x/projects/core/src/site-context/services/site-context-url-serializer.ts#L25) (for injecting site-context params into the prefix of the URL path, i.e. /electronics-spa/en/USD/... ).
